main.go

```diff
@@ -152,14 +152,26 @@ func run(ctx context.Context) {
 }
 
 func runLoop(ctx context.Context) {
+	go func() {
+		metadataPollTicker := time.NewTicker(config.MetadataPollInterval())
+		for {
+			logger.Debugf("Polling metadata.")
+			if err := config.SetConfig(ctx); err != nil {
+				logger.Errorf(err.Error())
+			}
+			select {
+			case <-metadataPollTicker.C:
+				continue
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+
 	var taskNotificationClient *agentendpoint.Client
 	var err error
 	ticker := time.NewTicker(config.SvcPollInterval())
 	for {
-		if err := config.SetConfig(ctx); err != nil {
-			logger.Errorf(err.Error())
-		}
-
 		if _, err := os.Stat(config.RestartFile()); err == nil {
```

Comment:
This restart check as well as the following config.TaskNotificationEnabled() checks should be able to be moved as well.

Reply:
i think its better to keep the agent restart tied to pollinterval since that can be controlled through metadata.